pkg/helm/release/manager.go

```diff
@@ -289,6 +293,17 @@ func generatePatch(existing, expected runtime.Object) ([]byte, error) {
 		return nil, nil
 	}
 
+	// We should add "remove" operations if the same path does also have
+	// a "add" operation
+	for _, rop := range removeOps {
+		for _, pop := range patchOps {
+			if rop.Path == pop.Path && pop.Operation == "add" {
+				patchOps = append(patchOps, rop)
+				break
+			}
+		}
+	}
+
```

Comment:
It was added after :

```
// If there are no patch operations, return nil. Callers are expected
	// to check for a nil response and skip the patch operation to avoid
	// unnecessary chatter with the API server.
	if len(patchOps) == 0 {
		return nil, nil
	}
``` 

Should not be before?

Reply:
If the patchOps slice is empty, the next for-loop wont change it anyway, so we can already return here

Reply:
Am I understanding correctly that it seems like the new library correctly chooses `replace` rather than adding both an `add` and `remove`?

Does that just mean this PR becomes just the dependency change to use the new library?